app/forms/ArrivalDetails.scala

```diff
@@ -16,17 +16,23 @@
 
 package forms
 
-import java.time.ZoneId
+import java.time.{ZoneId, ZonedDateTime}
 
 import forms.common.{Date, Time}
 import play.api.data.{Forms, Mapping}
 import play.api.libs.json.{Json, OFormat}
 
-case class ArrivalDetails(dateOfArrival: Date, timeOfArrival: Time)
+case class ArrivalDetails(dateOfArrival: Date, timeOfArrival: Time) {
+  def goodsArrivalMoment(zoneId: ZoneId): ZonedDateTime =
```

Comment:
I'll change the name to something different than Moment. It's not enough explanatory for me.